src/components/Table/GridTable.tsx

```diff
@@ -405,25 +410,29 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     // Split out the header rows from the data rows so that we can put an `infoMessage` in between them (if needed).
     const headerRows: RowTuple<R>[] = [];
     const totalsRows: RowTuple<R>[] = [];
-    const filteredRows: RowTuple<R>[] = [];
+    const visibleDataRows: RowTuple<R>[] = [];
+    const matchedRowIds: string[] = [];
```

Comment:
Nit/thought: Maybe call this `filteredRowIds`? I like the new `visibleDataRows` is "filtered && not collapsed".

...or maybe the name is good, and just in-source comment that this means something like "rows either explicitly matched by a filter or implicitly matched by a parent that matches the filter".